community/procedure-compiler/processor/src/main/java/org/neo4j/tooling/procedure/visitors/ContextFieldVisitor.java

```diff
@@ -19,45 +19,56 @@
  */
 package org.neo4j.tooling.procedure.visitors;
 
-import org.neo4j.tooling.procedure.messages.CompilationMessage;
-import org.neo4j.tooling.procedure.messages.ContextFieldWarning;
-import org.neo4j.tooling.procedure.messages.FieldError;
-
 import java.util.Arrays;
 import java.util.LinkedHashSet;
+import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import javax.lang.model.element.Element;
 import javax.lang.model.element.Modifier;
 import javax.lang.model.element.VariableElement;
 import javax.lang.model.type.TypeMirror;
 import javax.lang.model.util.Elements;
 import javax.lang.model.util.SimpleElementVisitor8;
 import javax.lang.model.util.Types;
 
+import org.neo4j.graphdb.DependencyResolver;
 import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.kernel.api.KernelTransaction;
+import org.neo4j.kernel.api.security.SecurityContext;
+import org.neo4j.kernel.api.security.UserManager;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
 import org.neo4j.logging.Log;
 import org.neo4j.procedure.Context;
+import org.neo4j.procedure.ProcedureTransaction;
+import org.neo4j.procedure.TerminationGuard;
+import org.neo4j.tooling.procedure.ProcedureProcessor;
+import org.neo4j.tooling.procedure.messages.CompilationMessage;
+import org.neo4j.tooling.procedure.messages.ContextFieldError;
+import org.neo4j.tooling.procedure.messages.ContextFieldWarning;
 
 class ContextFieldVisitor extends SimpleElementVisitor8<Stream<CompilationMessage>,Void>
 {
-    private static final Set<Class<?>> SUPPORTED_TYPES =
-            new LinkedHashSet<>( Arrays.asList( GraphDatabaseService.class, Log.class ) );
+    private static final Set<String> SUPPORTED_TYPES = new LinkedHashSet<>(
+            Arrays.asList( GraphDatabaseService.class.getName(), Log.class.getName(), TerminationGuard.class.getName(),
+                    SecurityContext.class.getName(), ProcedureTransaction.class.getName() ) );
+    private static final Set<String> RESTRICTED_TYPES = new LinkedHashSet<>(
+            Arrays.asList( GraphDatabaseAPI.class.getName(), KernelTransaction.class.getName(),
+                    DependencyResolver.class.getName(), UserManager.class.getName(),
+                    // the following classes are not in the compiler classpath
+                    "org.neo4j.kernel.enterprise.api.security.EnterpriseAuthManager",
```

Comment:
Open question: should I pull `neo4j-security-enterprise` and `neo4j-enterprise-kernel` in provided scope instead of hardcoding the two types like this?
I think it's a bad idea, though, as the compiler would only work against procedure for Neo4j Enterprise.